cmd/node.go

```diff
@@ -151,13 +151,30 @@ func (node *Node) Start(exitChannels ...*chan struct{}) {
 	// size.
 	//
 	// FIXME: We should rewrite the code so that PrefixBlockHashToUtxoOperations is either removed or written
-	// to badger in such a way as to not require the use of PerformanceBadgerOptions. Seet he comment on
+	// to badger in such a way as to not require the use of PerformanceBadgerOptions. See the comment on
 	// dirtyHackUpdateDbOpts.
-	if node.Config.HyperSync {
-		opts = lib.DefaultBadgerOptions(dbDir)
-	} else {
+
+	// Check to see if this node has already been initialized with performance or default options.
+	// If so, we should continue to use those options.
+	// If not, it should be based on the sync type.
+	performanceOptions, err := lib.DbInitializedWithPerformanceOptions(node.Config.DataDirectory)
+
+	// If the db options haven't yet been saved, we should base the options on the sync type.
+	if os.IsNotExist(err) {
```

Comment:
Whenever we encounter an annoying bug like this I basically like to leave the post-mortem in a long comment next to the change. Here, I would explain that the reason why you need to save the options to disk is because Badger breaks if you (a) shutdown (b) without closing (c) and then re-open with *different* options.

The other reason why this is good is it gives us the confidence to delete hacks like this in the future, rather than being scared that we'll break something if we do.